corehq/apps/analytics/tasks.py

```diff
@@ -257,19 +183,34 @@ def _send_post_data(url, params, data, headers):
     return response
 
 
-def _get_user_hubspot_id(web_user):
+def _get_user_hubspot_id(web_user, retry_num=0):
+    """
+    Attempts to match a web_user with a hubspot vid if that web user's email is
+    in hubspot.
+    :param web_user: WebUser
+    :param retry_num: the number of the current retry attempt
+    :return: string or None
+    """
+    if retry_num > 0:
+        time.sleep(10)  # wait 10 seconds if this is another retry attempt
+
     api_key = settings.ANALYTICS_IDS.get('HUBSPOT_API_KEY', None)
     if api_key and hubspot_enabled_for_user(web_user):
-        req = requests.get(
-            "https://api.hubapi.com/contacts/v1/contact/email/{}/profile".format(
-                six.moves.urllib.parse.quote(web_user.username)
-            ),
-            params={'hapikey': api_key},
-        )
-        if req.status_code == 404:
-            return None
-        req.raise_for_status()
-        return req.json().get("vid", None)
+        try:
+            req = requests.get(
+                "https://api.hubapi.com/contacts/v1/contact/email/{}/profile".format(
+                    six.moves.urllib.parse.quote(web_user.username)
+                ),
+                params={'hapikey': api_key},
+            )
+            if req.status_code == 404:
+                return None
+            req.raise_for_status()
+        except (ConnectionError, requests.exceptions.HTTPError):
+            if retry_num <= MAX_API_RETRIES:
+                return _get_user_hubspot_id(web_user, retry_num + 1)
```

Comment:
Should this `raise` on too many retries?
```suggestion
                return _get_user_hubspot_id(web_user, retry_num + 1)
            raise
```